src/pubtools/_impl/tracing.py

```diff
@@ -155,7 +193,26 @@ def force_flush(self):
             self._processor.force_flush()
         log.info("Flush trace data into OTEL collectors")
 
-    @property
-    def provider(self):
-        """Trace provider"""
-        return self._provider
+
+class CLITracingWrapper(TracingWrapper):
```

Comment:
I guess the idea behind the naming here is:

- ServerTracingWrapper: expected to be used when pubtools is loaded from within a service, in which case otel libraries are expected to be installed
- CLITracingWrapper: expected to be used when pubtools is loaded from a command-line tool somebody is running manually, in which case otel libraries are not expected to be installed

If we really need separate classes though, I would prefer a more neutral naming like StubTracingWrapper or NoopTracingWrapper because there is absolutely nothing wrong with not having otel libraries installed in a service, or having otel libraries installed when running CLI tools.

I also think the server vs CLI delineation is not clear. For example, this library is also loaded from some tekton tasks running in Konflux.  Is that the "server" case since the task execution is managed by a service, or the "CLI" case since it's ultimately running a `pubtools-*` process the same as one would do at the command-line?

Either way though, I'm questioning whether we need a separate class in the first place.
OTEL tracing is a feature which can be toggled, so the existing class already supports "do nothing".
Surely there is a way to harness that and make this case fall into the same "do nothing" code paths, rather than having two different ways to do nothing.

It seems to me that:

- in `_reset`, `_enabled_trace` could be forced to `False` if OTEL libraries are not available. The rest of the function therefore won't do anything.
- in `instrument_func`, the code is already written to have no effect on the wrapped function if tracing is disabled.  So this is very close to already working.  The only problem I can pick out is that the line `tracer = trace.get_tracer(__name__)` probably has to be moved behind an `OPENTELEMETRY_AVAILABLE` check.

So it's very close to already working.